src/sentry/snuba/metrics/extraction.py

```diff
@@ -567,13 +568,40 @@ def combine(cls, *supported_by):
         )
 
 
+def should_use_on_demand_metrics_for_querying(organization: Organization, **kwargs) -> bool:
+    """Helper function to check if an organization can query an specific on-demand function"""
+    components = _extract_aggregate_components(kwargs["aggregate"])
+    if components is None:
+        return False
+    function, _ = components
```

Comment:
We tables w/ columns should we be checking `is_function` per column, and checking those against the list as well?